collect_app/src/main/java/org/odk/collect/android/activities/FormDownloadListActivity.java

```diff
@@ -191,11 +204,10 @@ private void init(Bundle savedInstanceState) {
 
         downloadButton = findViewById(R.id.add_button);
         downloadButton.setEnabled(listView.getCheckedItemCount() > 0);
-        downloadButton.setOnClickListener(new OnClickListener() {
-            @Override
-            public void onClick(View v) {
-                downloadSelectedFiles();
-            }
+        downloadButton.setOnClickListener(v -> {
+            ArrayList<FormDetails> filesToDownload = getFilesToDownload();
+            logDownloadAnalyticsEvent();
```

Comment:
How would you feel about moving all of the analytics logging code to the ViewModel here? Ideally `startFormDownload` would be a ViewModel method and the analytics would be encapsulated within there but right now we could move logDownloadAnalyticsEvent to the ViewModel and then all the analytics testing/concern could be hidden from the Activity (you might still want a test to check the method is called).

Reply:
Makes sense. 👍